openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,435 @@
+package managementcpusoverride
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	PluginName = "autoscaling.openshift.io/ManagementCPUsOverride"
+	// timeToWaitForCacheSync contains the time how long to wait for caches to be synchronize
+	timeToWaitForCacheSync = 10 * time.Second
+	// podWorkloadAnnotationPrefix contains the prefix for the workload annotation
+	podWorkloadAnnotationPrefix = "workload.openshift.io/"
```

Comment:
Let's narrow the prefix one more level.  `target.workload.openshift.io/`?  This immediately conflicts with `workload.openshift.io/allowed` below, right?

Reply:
Narrowing this prefix further so we do not burn the entire second level prefix `workload.openshift.io` is important.

> Since this is tech preview, we do have the potential to change it later

You don't really.  The meaning will be inconsistent and it will be extremely difficult to change API fields later.



> but we're well into the implementation of this elsewhere across dozens of PRs and as Artyom points out we have to make the name match everywhere.

It is much easier to change now than it will be to change it later.

Reply:
https://github.com/openshift/enhancements/pull/739 proposes new names.

Reply:
done